liota/lib/transports/mqtt.py

```diff
@@ -191,22 +193,35 @@ def connect_soc(self):
         # Set up TLS support
         if self.tls_conf:
 
+            if self.identity is None:
+                raise ValueError("Identity required to be set")
+
+            # Creating the tls context
+            if ssl is None:
```

Comment:
Why fail here rather than at 38?

Reply:
I followed the implementation in Paho MQTT client as this piece of code (creating context) is similar to Paho MQTT internal code:
https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/client.py#L26

I believe the reason is to complete the installation of the library even if "ssl" is missing and then raise valueError if in case the user configures tls/ssl.

Reply:
Ah yes. Because we're still supporting non-TLS connections. Thank you.